### ISSUE TITLE
IndexError: Out of bounds on buffer access (axis 1)

### ISSUE THREAD

**Issue by Joozty** [explosion/thinc-apple-ops#3]
Hi I tried to use this awesome package and I am getting this error. Not sure what it means, maybe you guys could help me? 

I should mention that my data is quite big and I am also using some SWAP space. Could this be the reason of this error?

```python
[2021-09-28 21:09:01,238] [INFO] Set up nlp object from config
[2021-09-28 21:09:01,500] [INFO] Pipeline: ['tok2vec', 'ner', 'sentencizer', 'entity_linker']
[2021-09-28 21:09:01,505] [INFO] Created vocabulary
[2021-09-28 21:09:01,505] [INFO] Finished initializing nlp object
Traceback (most recent call last):
  File "/Users/joozty/Documents/kolurbo/venv/bin/spacy", line 8, in <module>
    sys.exit(setup_cli())
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/spacy/cli/_util.py", line 69, in setup_cli
    command(prog_name=COMMAND)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/typer/main.py", line 500, in wrapper
    return callback(**use_params)  # type: ignore
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/spacy/cli/train.py", line 60, in train_cli
    nlp = init_nlp(config, use_gpu=use_gpu)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/spacy/training/initialize.py", line 84, in init_nlp
    nlp.initialize(lambda: train_corpus(nlp), sgd=optimizer)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/spacy/language.py", line 1272, in initialize
    proc.initialize(get_examples, nlp=self, **p_settings)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/spacy/pipeline/tok2vec.py", line 216, in initialize
    self.model.initialize(X=doc_sample)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/thinc/model.py", line 299, in initialize
    self.init(self, X=X, Y=Y)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/thinc/layers/chain.py", line 86, in init
    layer.initialize(X=curr_input, Y=Y)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/thinc/model.py", line 299, in initialize
    self.init(self, X=X, Y=Y)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/thinc/layers/chain.py", line 90, in init
    curr_input = layer.predict(curr_input)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/thinc/model.py", line 315, in predict
    return self._func(self, X, is_train=False)[0]
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/thinc/layers/concatenate.py", line 44, in forward
    Ys, callbacks = zip(*[layer(X, is_train=is_train) for layer in model.layers])
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/thinc/layers/concatenate.py", line 44, in <listcomp>
    Ys, callbacks = zip(*[layer(X, is_train=is_train) for layer in model.layers])
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/thinc/model.py", line 291, in __call__
    return self._func(self, X, is_train=is_train)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/spacy/ml/staticvectors.py", line 46, in forward
    vectors_data = model.ops.gemm(model.ops.as_contig(V[rows]), W, trans2=True)
  File "/Users/joozty/Documents/kolurbo/venv/lib/python3.9/site-packages/thinc_apple_ops/ops.py", line 25, in gemm
    C = blas.gemm(x, y, trans1=trans1, trans2=trans2)
  File "thinc_apple_ops/blas.pyx", line 37, in thinc_apple_ops.blas.gemm
  File "thinc_apple_ops/blas.pyx", line 53, in thinc_apple_ops.blas.gemm
IndexError: Out of bounds on buffer access (axis 1)
```

## Info about spaCy

- **spaCy version:** 3.1.3
- **Platform:** macOS-11.6-arm64-arm-64bit
- **Python version:** 3.9.7
- **Pipelines:** en_core_web_sm (3.1.0), en_core_web_md (3.1.0)

**Comment by adrianeboyd** [comment]
We've been using this package internally in our day-to-day development without problems, but there are two test failures in the spacy test suite that are related to empty docs, which would line up with this kind of error. It's still kind of unexpected, but maybe this is the something that the blis and numpy/openblas options handle differently. We will look into it.

**Comment by adrianeboyd** [comment]
Okay, this should be fixed in v0.0.4.
